src/OpenColorIO/Op.h

```diff
@@ -71,19 +74,11 @@ OCIO_NAMESPACE_ENTER
         OpCPU() {}
         virtual ~OpCPU() {}
 
-        virtual void apply(float * rgbaBuffer, long numPixels) const = 0;
+        // All the Ops assume float pointers (i.e. always float bit depths) except 
+        // the 1D LUT CPU Op where the finalization depends on input and output bit depths.
+        virtual void apply(const void * inImg, void * outImg, long numPixels) const = 0;
```

Comment:
Could we offer both apply methods? Keep the existing interface which processes a single array in place (as an overload), and add the new in/out array apply? The single array version could simply call to the in/out version with the one array for both args, since many instances of apply are now doing that in this PR anyway.

Reply:
Good point, I added it.